Lib/idlelib/help.py

```diff
@@ -176,31 +176,39 @@ def bind_events(self):
         shortcut = 'Command' if darwin else 'Control'
         # Zoom out works with or without shift.
         self.widget.event_add(
-                '<<zoom-text-out>>',
+                '<<zoom-text-in>>',
                 *[f'<{shortcut}-Key-equal>', f'<{shortcut}-Key-plus>'])
-        self.widget.bind('<<zoom-text-out>>', self.zoom)
+        self.widget.bind('<<zoom-text-in>>', self.zoom_in)
 
         self.widget.event_add(
-                '<<zoom-text-in>>',
+                '<<zoom-text-out>>',
                 *[f'<{shortcut}-Key-minus>', f'<{shortcut}-Key-underscore>'])
-        self.widget.bind('<<zoom-text-in>>', self.zoom)
+        self.widget.bind('<<zoom-text-out>>', self.zoom_out)
 
         # Windows and Mac use MouseWheel.
-        self.widget.bind('<Control-MouseWheel>', self.zoom)
+        self.widget.bind('<Control-MouseWheel>', self.zoom_mousewheel)
         # Linux uses Button 4 (scroll down) and Button 5 (scroll up).
-        self.widget.bind('<Control-Button-4>', self.zoom)
-        self.widget.bind('<Control-Button-5>', self.zoom)
-
-    def zoom(self, event):
-        "Handle zooming in/out."
-        if event.type == EventType.KeyPress:
-            increase = event.keysym in {'plus', 'equal'}
-        elif event.type == EventType.MouseWheel:
-            increase = event.delta >= 0 == darwin
-        elif event.type == EventType.Button:
-            increase = event.num == 4
+        self.widget.bind('<Control-Button-4>', self.zoom_mousewheel)
+        self.widget.bind('<Control-Button-5>', self.zoom_mousewheel)
+
+    def zoom_in(self, event):
+        "Make font size larger."
+        self.zoom(increase=True)
+
+    def zoom_out(self, event):
+        "Make font size smaller."
+        self.zoom(increase=False)
+
+    def zoom_mousewheel(self, event):
+        "Adjust font size based on mouse wheel direction."
+        wheel_up = {EventType.MouseWheel: event.delta > 0,
```

Comment:
I wasn't sure when that logic needed to be applied and when it didn't.  So, it's only adjusted on scrolling and not on other uses of the mousewheel?

Reply:
I tested turtledemo a month ago and font resizing works the same as on Windows.  Moving the top of the wheeI up increases size.  I don't actually know if this is the correct convention on Mac, but in the absence of information otherwise, including the absence of complaint about turtledemo, I will consider it correct.  I may have tested command-wheel-up in Safari and will recheck tomorrow.

When we copied to logic for scrolling, the result was buggy since the convention on Mac, which I know I verified with Safari, is the opposite of Windows (and Linux, I believe).  I will try to find and point you to the tracker discussion.